apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SqlResultStream.cs

```diff
@@ -203,8 +204,9 @@ internal class RetryState
             private readonly IScheduler _scheduler;
             private readonly RetrySettings _retrySettings;
             private readonly int _maxConsecutiveErrors;
-            private TimeSpan _nextBackoff;
-            // TODO: Possibly use RetrySettings.MaxAttempts instead of having this separately?
+            private IEnumerator<TimeSpan> _retrySettingsBackoffs;
+            // Note that we can't use RetrySettings.MaxAttempts, as our sequence of backoffs is not advanced
```

Comment:
I think my proposal above would solve this one as well if it's OK for next backoff to be calculated from the recommended delay, which I think it's OK because the server is probably sending the recommended delay always so we'll really never use the calculated backoffs.

Reply:
Yes, it would probably solve this one too. If we see the same thing in *other* APIs, I'd be happy to add it to GAX at that point.